client/src/legacy/TinyMCE_sslink-internal.js

```diff
@@ -119,7 +119,17 @@ jQuery.entwine('ss', ($) => {
 
     getOriginalAttributes() {
       const editor = this.getElement().getEditor();
-      const node = $(editor.getSelectedNode());
+
+      // Find "a" node, issue https://github.com/silverstripe/silverstripe-cms/issues/2439
+      let linkNode = editor.getSelectedNode();
+      if (linkNode.nodeName !== 'A') {
+        let count = 0;
+        while (count < 10 && linkNode.parentNode && linkNode.nodeName !== 'A') {
+          linkNode = linkNode.parentNode;
+          count +=1;
+        }
+      }
```

Comment:
Please don't use a while loop - especially with a completely arbitrary number of iterations. If there's a function available to recursively check the ancestors of the element, use that. Otherwise, this might need to go back to the drawing board.